src/v/cluster/health_monitor_backend.cc

```diff
@@ -789,6 +790,62 @@ health_monitor_backend::get_node_drain_status(
     co_return it->second.drain_status;
 }
 
+health_monitor_backend::aggregated_report
+health_monitor_backend::aggregate_reports(report_cache_t& reports) {
+    struct collector {
+        absl::node_hash_set<model::ntp> to_ntp_set() const {
+            absl::node_hash_set<model::ntp> ret;
+            for (const auto& [topic, parts] : t_to_p) {
+                for (auto part : parts) {
+                    ret.emplace(topic.ns, topic.tp, part);
+                    if (
+                      ret.size() == aggregated_report::max_partitions_report) {
+                        return ret;
+                    }
+                }
+            }
+            return ret;
+        }
+
+        size_t count() const {
+            size_t sum = 0;
+            for (const auto& [_, parts] : t_to_p) {
+                sum += parts.size();
+            }
+            return sum;
+        }
+
+        absl::node_hash_map<
+          model::topic_namespace,
+          absl::node_hash_set<model::partition_id>>
```

Comment:
Might be overthinking this but did you try making this a bitset/vector<uint8_t>. The partition ids should be strictly incrementing from 0? The abseil maps are quite expensive for basic incrementing integers because of all the hashing.

Reply:
`std::bitset` is statically sized so it wouldn't be suitable here but yes a "bitmap" is the ideal representation I think for partitions. We should have classes like "set of ntps" which is similar in principle to set<ntp> or vector<ntp> but internally stores the topic (for each topic) attached to a bitmap of the partitions.

That's a longer term thing though that I need needs some careful design and I didn't want to do it here.

> The abseil maps are quite expensive for basic incrementing integers because of all the hashing.

Also (perhaps mostly?) due to all the allocation too. Node hash map has at least 1 allocation per insert and also allocates sometimes on failed inserts (i.e., element already exists).

Reply:
> The partition ids should be strictly incrementing from 0?

Not sure I understand the question, but the IDs are "dense": a topic with N partitions has partition IDs exactly in the range [0, N).

Reply:
> That's a longer term thing though that I need needs some careful design and I didn't want to do it here.

Fair enough was just wondering given you already looked into perf for this one. Bit surprised we don't have something like that yet.

 > Not sure I understand the question, but the IDs are "dense": a topic with N partitions has partition IDs exactly in the range [0, N).

Yep was wondering whether there is possibly some weird edge case but sounds like not so the use of a better datastructure is definitely possible.

Reply:
> Fair enough was just wondering given you already looked into perf for this one. Bit surprised we don't have something like that yet.

Right, this was on the TODO for me at least sicne the fetch planning investgiation where it also would have been useful. H/e I think it to do it right with say CRoaring is a fair amount of work and probably not a good backport target. This is a backport target.

> Yep was wondering whether there is possibly some weird edge case but sounds like not so the use of a better datastructure is definitely possible.

Even if they weren't dense for some reason a "bitmap" should still work and with CRoaring sparse bitmaps or bitmaps with all their bits offset far away from 0 index are handled efficiently.